### ISSUE TITLE
Support for cluster volumes

### ISSUE THREAD

**Issue by TheSilkky** [aaraney/deployx#2]
I noticed when trying to deploy a stack that uses cluster volumes, deployx will output this error:
`service name_of_service: volume type must be volume, bind, tmpfs, or npipe`

Cluster volumes are a part of swarm and should be supported by deployx, however they aren't apart of the compose-spec which might be an issue

[Cluster Volumes (CSI) docs](https://github.com/moby/moby/blob/master/docs/cluster_volumes.md)

Example:
```
postgres:
  image: postgres:15.3-alpine3.18
  hostname: postgres
  networks:
    - backend
  volumes:
    - type: cluster
      source: postgres_database
      target: /var/lib/postgresql/data
  secrets:
    - source: database_password
      mode: 0600
  environment:
    - POSTGRES_DB=example
    - POSTGRES_USER=example
    - POSTGRES_PASSWORD_FILE=/run/secrets/database_password
    - PGDATA=/var/lib/postgresql/data/pgdata
```

**Comment by aaraney** [comment]
Thanks for reporting this, @TheSilkky! Im not sure this is something that `deployx` will _add_ support for ourselves. Instead when / if compose-spec (i.e. `compose-go`) adds the `cluster` volume type as an option, we will certainly pull in that change. Ill open an issue in `compose-go` to see if they are willing to pull in the addition and ref that here. I agree with you though, this is something that compose-spec should support.

**Comment by aaraney** [comment]
I may have spoken too soon, @TheSilkky. This does look like something `deployx` can add ourselves without incurring any friction with `compose-go`. I would like to get the `cluster` volume type constant into `compose-go`, but `deployx` will need to implement the feature. Ill start to have a look at this now. Would you be willing to pull down a branch once I have something up and test it out?

**Comment by TheSilkky** [comment]
Sure, I'll test it out when your ready

**Comment by aaraney** [comment]
@TheSilkky, I pushed up an initial draft version to a [`add-cluster-volume-support`](https://github.com/aaraney/deployx/tree/add-cluster-volume-support) branch. If you could test that out locally that would be great! See the [_building from source_](https://github.com/aaraney/deployx#building-from-source) section of the readme for support on building the binary.

**Comment by aaraney** [comment]
@TheSilkky, I just verified that it works on my machine using `democratic-csi`'s `local-hostpath` driver.

**Comment by aaraney** [comment]
@TheSilkky, I released a new version, v0.0.2, that adds support for cluster volumes and mentioned you in the changelog! Thanks again for the suggestion!
